lib/Doctrine/Annotations/DocParser.php

```diff
@@ -443,13 +442,8 @@ private function classExists($fqcn)
             return $this->classExists[$fqcn];
         }
 
-        // first check if the class already exists, maybe loaded through another AnnotationReader
-        if (class_exists($fqcn, false)) {
-            return $this->classExists[$fqcn] = true;
-        }
-
         // final check, does this class exist?
-        return $this->classExists[$fqcn] = AnnotationRegistry::loadAnnotationClass($fqcn);
+        return $this->classExists[$fqcn] = class_exists($fqcn);
```

Comment:
The entire private method can be dropped

Reply:
If not: what is our reason for caching a failure?

Reply:
This whole thing will be gone soon once Hoa parser is in place so I'd ignore that for now.

Reply:
Alright, let's not touch it for now 👌

Reply:
> This whole thing will be gone soon once Hoa parser is in place

What's that?

Reply:
Surprise. :gift: